r/tools/nixlibs.R

```diff
@@ -539,10 +562,25 @@ build_libarrow <- function(src_dir, dst_dir) {
     env_var_list <- c(env_var_list, ARROW_DEPENDENCY_SOURCE = "BUNDLED")
   }
 
+  # On macOS, if not otherwise set, let's override Boost_SOURCE to be bundled
+  if (on_macos) {
+    deps_to_bundle <- c("Boost", "lz4")
+    for (dep_to_bundle in deps_to_bundle) {
+      env_var <- paste0(dep_to_bundle, "_SOURCE")
+      if (Sys.getenv(env_var) == "") {
```

Comment:
We might _also_ want to check `Sys.getenv(toupper(env_var))` here in case someone also has the upper variant laying around. But this should be very rare (and only really developers)

Reply:
No harm in adding it just in case?

Reply:
\<Dependency\>_SOURCE is actually a CMake only variable outside of the few variants we use in `build_arrow_static.sh` (a fact that cost me 2 hours debugging a workflow change once xD). So it should be fine to only check the _right_ version.

Reply:
Given kou's comment (and my incoming changes to use the mix-case version) I think I'll keep this as mixed case to match the (current) cmake args. I probably should also add this to the documentation of our build variables in the package